addons/vehicles/CfgVehicles/Viper.inc

```diff
@@ -0,0 +1,191 @@
+#define CAMO_NET_TEXTURE QUOTE(A3\Armor_F\Data\camonet_AAF_Digi_Green_CO.paa)
+
+class SWS_V_MBT_M700_Viper : I_MBT_03_cannon_F {
+    ITEM_META(2);
+    displayName = QNAME(M700 Viper);
+    faction = "SWS_UNSC";
+    side = 1;
+    crew = "OPTRE_UNSC_Marine_Soldier_Crewman";
+    typicalCargo[] = { "OPTRE_UNSC_Marine_Soldier" };
+    editorPreview = QPATHTOF(data\viper\editor_preview.jpg);
+    
+    radarType=2;
+    reportRemoteTargets=1;
+    reportOwnPosition=1;
+    receiveRemoteTargets=1;
+    class Components: Components {
+        class SensorsManagerComponent {
+            class Components {
+                class DataLinkSensorComponent: SensorTemplateDataLink { };
+                class PassiveRadarSensorComponent: SensorTemplatePassiveRadar {
+                    class AirTarget {
+                        minRange=8000;
+                        maxRange=8000;
+                        objectDistanceLimitCoef=-1;
+                        viewDistanceLimitCoef=-1;
+                    };
+                    class GroundTarget {
+                        minRange=8000;
+                        maxRange=8000;
+                        objectDistanceLimitCoef=-1;
+                        viewDistanceLimitCoef=-1;
+                    };
+                };
+            };
+        };
+    };
+
+    class Turrets: Turrets {
+        class MainTurret: MainTurret {
+            class Turrets: Turrets {
+                class CommanderOptics: CommanderOptics {
+                    weapons[] = {
+                        "HMG_127_APC",
+                        "Laserdesignator_mounted",
+                        "SmokeLauncher"
+                    };
+                    magazines[] = {
+                        "200Rnd_127x99_mag_Tracer_Yellow",
+                        "200Rnd_127x99_mag_Tracer_Yellow",
+                        "200Rnd_127x99_mag_Tracer_Yellow",
+                        "SmokeLauncherMag",
+                        "SmokeLauncherMag",
+                        "SmokeLauncherMag",
+                        "SmokeLauncherMag",
+                        "SmokeLauncherMag",
+                        "Laserbatteries"
+                    };
+                    class Components {
+                        class VehicleSystemsDisplayManagerComponentLeft: VehicleSystemsTemplateLeftCommander {
+                            class Components: components {
```

Comment:
```suggestion
                            class Components: Components {
```

Reply:
Lowercase here is right--that's how `VehicleSystemsTemplateLeftCommander` is defined

Reply:
That's not accurate; your base class is `class Components;` in your `CfgVehicles.inc` so you should have this uppercased. hemtt will warn you if you run a build with the latest version.

Reply:
Okay, it took me a second to verify and sense-make this for myself, but I think I understand what's happening now.

In their source design for the T-14K, BI defined the `VehicleSystemsTemplate` classes with lowercased `components` members. I'm... not sure why they did this but I'd imagine it's something to do with the way they were structuring their inheritance. With the global config system we're using, you're right in that it should be capitalized. Thanks for checking me on that :>